### PR TITLE
**BREAKING**: Fetch and return token image as part of `getDetails` calls on ERC721Standard and ERC1155Standard

### LINKED ISSUE #689: getDetails method in ERC721Standard and ERC1155Standard classes should return image/description instead of TokenURI
The `getDetails` methods for `ERC721Standard` and `ERC1155Standard` currently fetch and return a tokenURI. Ideally they would  fetch the tokenURI and then use it to fetch other more useful details like `image` `name` and `description` (possibly more) as we do in the `getCollectibleInformationFromTokenURI` method of the `CollectiblesController`.

### PR DESCRIPTION
**Pull request by adonesky1**
- ADDED:

  - **BREAKING** Adds functionality to fetch and return token images as part of `getDetails` calls on ERC721Standard and ERC1155Standard
     - This change is breaking because it requires that the AssetsContractController (on which the ERC721Standard and ERC1155Standard are instantiated) be passed a listener for `onPreferencesStateChange` from the `PreferencesController` so that it can use the user's preferred IPFSGateway to fetch any images hosted on IPFS. Consumers will have to pass `onPreferencesStateChange` in an options object (first arg) to the AssetsContractController constructor when initializing.

Resolves #689

### CODE REVIEW

**Review by wachunei — APPROVED** [review]
LGTM

**Review by mcmire — CHANGES_REQUESTED** [review]
Two questions:

1. Are the new tests for `getDetails` actually hitting the IPFS gateway? I see you were trying to use Nock to mock some kind of API, perhaps it would be worth doing this for the IPFS gateway? This raises the question of "how should we handle external APIs in tests?" again.
2. Perhaps tying into 1, is it possible to test the case in which `getDetails` returns a non-nullish value for `image` in the expected set of data?
3. How are we testing the new change to AssetsContractController where `onPreferencesStateChange` can update the configuration object?

**Comment by adonesky1** [comment]
> Three questions:

All good callouts. Will add/modify tests to address these concerns.

File: src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.test.ts [hunk 1]
```diff
@@ -309,7 +346,134 @@ describe('ERC721Standard', () => {
...
+            'https://bafybeie5ycnlx5ukzhlecakrbjeqnkpanuolobatuqooaeguptulganq6u.ipfs.cloudflare-ipfs.com',
+        };
+      });
+
+    const expectedResult = {
+      name: 'BoredApeYachtClub',
+      standard: 'ERC721',
+      symbol: 'BAYC',
+      tokenURI:
+        'https://bafybeihpjhkeuiq3k6nqa3fkgeigeri7iebtrsuyuey5y6vy36n345xmbi.ipfs.cloudflare-ipfs.com/3',
```
Comment (adonesky1) [inline src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.test.ts:440]: tokenURI hosted on IPFS cc @mcmire @wachunei

File: src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.test.ts [hunk 2]
```diff
@@ -276,25 +280,58 @@ describe('ERC721Standard', () => {
...
+        return {
+          image: 'https://card.godsunchained.com/?id=1329&q=4',
+        };
+      });
+
     const expectedResult = {
       name: 'Gods Unchained',
       standard: 'ERC721',
       symbol: 'GODS',
       tokenURI: 'https://api.godsunchained.com/card/4',
```
Comment (adonesky1) [inline src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.test.ts:295]: tokenURI not hosted on IPFS cc @mcmire @wachunei

File: src/assets/AssetsContractController.test.ts [hunk 3]
```diff
@@ -16,12 +18,30 @@ const TEST_ACCOUNT_PUBLIC_ADDRESS =
...
+
+  it('should update the ipfsGateWay config value when this value is changed in the preferences controller', () => {
+    expect(assetsContract.config).toStrictEqual({
+      ipfsGateway: IPFS_DEFAULT_GATEWAY_URL,
+      provider: undefined,
+    });
+
+    preferences.setIpfsGateway('newIPFSGateWay');
+    expect(assetsContract.config).toStrictEqual({
+      ipfsGateway: 'newIPFSGateWay',
```
Comment (adonesky1) [inline src/assets/AssetsContractController.test.ts:44]: test new preferences listener updates to controller config values cc @mcmire

**Comment by adonesky1** [comment]
>Thanks for updating these tests. For these types of tests where you've modified how the controller is instantiated so that you now pass onPreferencesStateChange, does this have any effect on the tests? For instance, if onPreferencesStateChange is required, could you get away with saying this?

For all except the [test I added on `AssetsContractController.test.ts`](https://github.com/MetaMask/controllers/pull/702#discussion_r819009973) you could pass 
```
onPreferencesStateChange: (listener) => {
    // do nothing
  }
```

**Review by mcmire — APPROVED** [review]

**Review by wachunei — APPROVED** [review]

**Pull request merged**